ssl/statem/extensions_clnt.c

```diff
@@ -678,12 +678,41 @@ EXT_RETURN tls_construct_ctos_early_data(SSL *s, WPACKET *pkt,
                                          unsigned int context, X509 *x,
                                          size_t chainidx, int *al)
 {
+    const unsigned char *id;
+    size_t idlen;
+    SSL_SESSION *psksess = NULL;
+    const EVP_MD *handmd = NULL;
+
+    if (s->hello_retry_request)
+        handmd = ssl_handshake_md(s);
```

Comment:
`s->early_data_state != SSL_EARLY_DATA_CONNECTING` in the event of an HRR so we will hit the `return EXT_NOT_SENT` below in that case. However it looks like HRR with early_data is screwed anyway. Fix in #3933. Probably I need to hold off with this PR until that has gone in, and then rebase this one.